stars-core/src/main/kotlin/tools/aqua/stars/core/hooks/defaulthooks/EmptyTSCHook.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 The STARS Project Authors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tools.aqua.stars.core.hooks.defaulthooks
+
+import tools.aqua.stars.core.hooks.EvaluationHookResult
+import tools.aqua.stars.core.hooks.PreEvaluationHook
+import tools.aqua.stars.core.types.*
+
+/** [PreEvaluationHook] that checks if a [TSC] is empty, i.e. has no nodes. */
+open class EmptyTSCHook<
+    E : EntityType<E, T, S, U, D>,
+    T : TickDataType<E, T, S, U, D>,
+    S : SegmentType<E, T, S, U, D>,
+    U : TickUnit<U, D>,
+    D : TickDifference<D>> :
+    PreEvaluationHook<E, T, S, U, D>(
+        identifier = "EmptyTSCHook",
+        evaluationFunction = { tsc ->
+          if (tsc.toList().size > 1) EvaluationHookResult.OK else EvaluationHookResult.SKIP
```

Comment:
Shouldn't it be `tsc.toList().size > 0`? In the failing "TSCMonitorTest" there is a node in the TSC. I would argue that the manually added "root" node is already a "TSC".

Reply:
Fixed.